lib/prepare-driver.js

```diff
@@ -16,53 +16,67 @@ module.exports = function (app){
     var adapterModuleName = adapterDef.moduleName;
     var pathToAdapterDependency = adapterDef.modulePath;
     var adapterConfig = adapterDef.config;
+    var adapterClientName = adapterConfig.adapterClientName || 'redis';
+    var adapterClientConfig = adapterConfig.adapterClientConfig || {};
 
     // For redis:
     // ============================================================
     if (adapterModuleName === 'socket.io-redis') {
 
       // Create raw redis clients if necessary
-      var rawClientsNecessary = adapterConfig.pass || adapterConfig.db || adapterDef.adminBus;
+      var rawClientsNecessary = adapterConfig.pass || adapterConfig.db || adapterDef.adminBus || adapterConfig.adapterClientConfig;
       if (!rawClientsNecessary) {
         return cb();
       }
 
-      // Borrow access to the `redis` module from socket.io-redis
+      // Borrow access to the adapter redis client module from socket.io-redis
       var redis;
       try {
-        redis = require(path.resolve(pathToAdapterDependency, 'node_modules/redis'));
+        redis = require(adapterClientName);
       }
       catch (e) {
         try {
-          redis = require('redis');
+          redis = require(path.resolve(pathToAdapterDependency, 'node_modules/redis'));
         } catch (e2) {
-          return cb(e2);
+          try {
+            redis = require('redis');
+          } catch (e3) {
+            return cb(e3);
+          }
         }
       }
 
-      // Set up object that will be used for redis client options
-      var redisClientOpts = {};
-
       // If `pass` was supplied, pass it in as `auth_pass`
       if (adapterConfig.pass) {
-
-        redisClientOpts.auth_pass = adapterConfig.pass;
+        adapterClientConfig.auth_pass = adapterConfig.pass;
       }
 
-
+      var initiateRedisClient = function(adapterConfig, redis, adapterClientName, adapterClientConfig){
```

Comment:
I know the Sails code base hasn't always been consistent about this in the past, but we actively try to avoid inline function declarations like this.  In this case, it's a 2x duplication, so I would just inline the code.  If this happened more than that, then I would pull it into another file and `require` that.